src/main/groovy/io/micronaut/build/MicronautPublishingPlugin.groovy

```diff
@@ -63,126 +67,135 @@ class MicronautPublishingPlugin implements Plugin<Project> {
+                        from project.sourceSets.main.allJava
+                    }
 
-                repositories {
-                    maven {
-                        credentials {
-                            username = bintrayUser
-                            password = bintrayKey
-                        }
-                        url  "https://oss.jfrog.org/oss-snapshot-local"
+                    project.task('javadocJar', type:Jar) {
+                        archiveClassifier.set('javadoc')
+                        from project.javadoc.destinationDir
                     }
                 }
 
-                publications {
-                    maven(MavenPublication) { publication ->
-                        artifactId( "micronaut-" + project.name.substring(project.name.indexOf('/') + 1) )
-                        def shadowJar = tasks.findByName("shadowJar")
-                        if(shadowJar) {
-                            artifact(project.tasks.shadowJar) {
-                                classifier = null
+                publishing {
+
+                    repositories {
+                        maven {
+                            credentials {
+                                username = bintrayUser
+                                password = bintrayKey
                             }
-                            pom.withXml { xml ->
-                                def xmlNode = xml.asNode()
-                                def dependenciesNode = xmlNode.appendNode('dependencies')
-
-                                project.configurations.shadow.allDependencies.each {
-                                    if (! (it instanceof SelfResolvingDependency)) {
-                                        def dependencyNode = dependenciesNode.appendNode('dependency')
-                                        dependencyNode.appendNode('groupId', it.group)
-                                        if (it.group.startsWith('io.micronaut')) {
-                                            dependencyNode.appendNode('artifactId', "micronaut-$it.name")
-                                        } else {
-                                            dependencyNode.appendNode('artifactId', it.name)
+                            url  "https://oss.jfrog.org/oss-snapshot-local"
+                        }
+                    }
+
+                    publications {
+                        if (project.extensions.findByType(PublishingExtension).publications.empty) {
+                            maven(MavenPublication) { publication ->
+                                artifactId( "micronaut-" + project.name.substring(project.name.indexOf('/') + 1) )
+
+                                if(project.shadowJarEnabled == true) {
+                                    def shadowJar = tasks.findByName("shadowJar")
+                                    artifact(project.tasks.shadowJar) {
+                                        classifier = null
```

Comment:
I believe `classifier` is deprecated and you have to use `artefactClassifier.set(null)` in newer versions of Gradle

Reply:
It is not deprecated nor it produces any deprecation warning.

As far as I can tell, `classifier = null` will invoke [this method](https://docs.gradle.org/current/javadoc/org/gradle/api/publish/maven/MavenArtifact.html#setClassifier-java.lang.String-) which is not deprecated.